.github/actions/npm_install_from_cache/action.yml

```diff
@@ -0,0 +1,15 @@
+name: npm_install_from_cache
+description: Install npm packages from cache
+runs:
+  using: composite
+  steps:
+    - name: Get cached dependencies
+      id: cache-npm
+      uses: actions/cache/restore@e12d46a63a90f2fae62d114769bbf2a179198b5c
+      with:
+          path: node_modules
+          key: npm-${{ hashFiles('./package-lock.json') }}
```

Comment:
typically runner.os is also added (although I doubt the lack of it will ever actually generate problems in this case)
key: ${{ runner.os }}-build-master-cache-${{ hashFiles('./package-lock.json') }}

Reply:
true, its optional as far as i know